sktime/lagged_model_validator.py

```diff
@@ -0,0 +1,276 @@
+
+    @property
+    def predictions_conf(self):
+        """ Returns the confidence intervals of the estimates at different
+        lagtimes (if available)
+
+        If not available, returns None.
+
+        Returns
+        -------
+        L : ndarray(T, n)
+            each row contains the lower confidence bound of n observables
+            computed at one of the T lag times.
+
+        R : ndarray(T, n)
+            each row contains the upper confidence bound of n observables
+            computed at one of the T lag times.
+
+        """
+        return self._predictions_conf
+
+
+class LaggedModelValidator(Estimator, metaclass=abc.ABCMeta):
+    r""" Validates a model estimated at lag time tau by testing its predictions
+    for longer lag times
+
+    Parameters
+    ----------
+    test_model : Model
+        Model to be tested
+
+    test_estimator : Estimator
+        Parametrized Estimator that has produced the model
+
+    mlags : int or int-array, default=10
+        multiples of lag times for testing the Model, e.g. range(10).
+        A single int will trigger a range, i.e. mlags=10 maps to
+        mlags=range(1, 10). The setting None will choose mlags automatically
+        according to the longest available trajectory
+
+    conf : float, default = 0.95
+        confidence interval for errors
+
+    err_est : bool, default=False
+        if the Estimator is capable of error calculation, will compute
+        errors for each tau estimate. This option can be computationally
+        expensive.
+
+    """
+    def __init__(self, test_model, test_estimator, mlags=None, conf=0.95, err_est=False):
+        # set model and estimator
+        # TODO: subclass cktest takes a copy as well, how to avoid two copies while being general?
+        self.test_model = test_model.copy()
+        import copy
+        self.test_estimator = copy.deepcopy(test_estimator)
+        # set conf and error handling
+        self.conf = conf
+        self.has_errors = hasattr(test_model, 'samples')
+        self.mlags = mlags
+
+        self.err_est = err_est
+        if err_est and not self.has_errors:
+            raise ValueError('Requested errors on the estimated models, '
+                             'but the model is not able to calculate errors at all')
+
+        super(LaggedModelValidator, self).__init__()
+
+    def _set_mlags(self, data, lagtime: int):
+        # set mlags, we do it in fit, so we can obtain the maximum possible lagtime from the trajectory data.
+        from numbers import Integral
+        if not isinstance(data, list):
+            data = [data]
+
+        mlags = self.mlags
+        maxlength = np.max([len(x) for x in data])
+        maxmlag = int(np.floor(maxlength / lagtime))
+        if mlags is None:
+            mlags = maxmlag
+        elif isinstance(mlags, Integral):
+            mlags = np.arange(mlags)
+        mlags = np.asarray(mlags, dtype='i')
+        if np.any(mlags > maxmlag):
```

Comment:
i would warn or raise in this case as its an invalid argument

Reply:
there used to be a warning, I will restore it.